### PR TITLE
refactor delete to remove cobra dependency

### PR DESCRIPTION
**Pull request by adohe-zz**
FYI. As part of CLI Q1 roadmap, we would like to reduce the dependency of Cobra from actual commands implementations. In this PR, I tried to refactor delete command to achieve this. @kubernetes/sig-cli-misc a quick review is quite welcome, and I am just working on more PRs.

### CODE REVIEW

**Comment by AdoHe** [comment]
@ymqytw I just notice this, bot is not smart enough.

**Comment by AdoHe** [comment]
@kubernetes/sig-cli-misc could someone give a quick review? I would like to move this forward.

File: pkg/kubectl/cmd/delete.go [hunk 1]
```diff
@@ -110,92 +135,109 @@ func NewCmdDelete(f cmdutil.Factory, out, errOut io.Writer) *cobra.Command {
...
+	cmd.Flags().BoolVar(&options.IgnoreNotFound, "ignore-not-found", false, "Treat \"resource not found\" as a successful delete. Defaults to \"true\" when --all is specified.")
+	cmd.Flags().BoolVar(&options.Cascade, "cascade", true, "If true, cascade the deletion of the resources managed by this resource (e.g. Pods created by a ReplicationController).  Default true.")
+	cmd.Flags().IntVar(&options.GracePeriod, "grace-period", -1, "Period of time in seconds given to the resource to terminate gracefully. Ignored if negative.")
+	cmd.Flags().BoolVar(&options.DeleteNow, "now", false, "If true, resources are signaled for immediate shutdown (same as --grace-period=1).")
+	cmd.Flags().BoolVar(&options.ForceDeletion, "force", false, "Immediate deletion of some resources may result in inconsistency or data loss and requires confirmation.")
+	cmd.Flags().DurationVar(&options.Timeout, "timeout", 0, "The length of time to wait before giving up on a delete, zero means determine a timeout from the size of the object")
+	cmd.Flags().StringVarP(&options.Output, "output", "o", "", "Output mode. Use \"-o name\" for shorter output (resource/name).")
```
Comment (fabianofranz) [inline pkg/kubectl/cmd/delete.go]: Can you add `AddOutputVarFlagsForMutation` for this, the same way you did `AddInclude3rdPartyVarFlags`?

**Review by fabianofranz — CHANGES_REQUESTED** [review]
One minor nit otherwise LGTM.

File: pkg/kubectl/cmd/delete.go (hunk 1 above)
Comment (AdoHe) [inline pkg/kubectl/cmd/delete.go]: sure.

Comment (AdoHe) [inline pkg/kubectl/cmd/delete.go]: Done. @fabianofranz could you please help review this PR? The bot just assign myself as the reviewer.

**Comment by AdoHe** [comment]
@k8s-bot gci gce e2e test

**Comment by fabianofranz** [comment]
@k8s-bot gci gce e2e test this

**Review by fabianofranz — APPROVED** [review]

**Pull request merged**